src/NHibernate/Driver/SqlServerCeDriver.cs

```diff
@@ -102,10 +98,6 @@ protected override void InitializeParameter(DbParameter dbParam, string name, Sq
 			base.InitializeParameter(dbParam, name, AdjustSqlType(sqlType));
 
 			AdjustDbParamTypeForLargeObjects(dbParam, sqlType);
-			if (prepareSql)
-			{
-				SqlClientDriver.SetVariableLengthParameterSize(dbParam, sqlType);
-		}
```

Comment:
What do you think of this removal? It works but is in contradiction with [documentation](https://msdn.microsoft.com/en-us/library/system.data.sqlserverce.sqlcecommand.prepare(v=vs.100).aspx#Anchor_2):

> Before you call Prepare, specify the data type of each parameter in the statement to be prepared. For each parameter that has a variable-length data type, you must set the Size property to the maximum size needed. Prepare returns an error if these conditions are not met.

Reply:
It is strange that it works.

Reply:
I dislike it quite. But that is the simplest change to get it work... Otherwise I could move the code from `SqlClientDriver` into Sql CE driver and adjust it to its needs, meaning removing 2008 stuff and not specifying blob sizes, but letting all the other been set, since it will be more compliant with doc and should work too.

Reply:
I don't think that SqlServer CE requires having all parameter sizes to be set when `Prepare` is used.

Reply:
So we just keep that change as is. This PR could then be merged after dropping the first commit.